cherab/tools/observers/bolometry.py

```diff
@@ -687,6 +695,350 @@ def etendue_single_run(_):
+    oriented along the observer's z axis.
+
+    :param str detector_id: The name for this detector.
+    :param Point3D centre_point: The centre point of the detector.
+    :param Vector3D basis_x: The x basis vector for the detector.
+    :param float width: The width of the detector along the x basis vector.
+    :param tuple pixels: The number of pixels to divide the foil into.
+      Pixels are square, so the height of the foil is determined by the
+      width of the foil and the number of rows and columns of pixels.
+    :param Node parent: The parent scenegraph node to which this detector belongs.
+      Typically a BolometerCamera() or an optical World() object.
+    :param bool units: The units in which to perform observations, can
+      be ['Power', 'Radiance'], defaults to 'Power'.
+    :param bool accumulate: Whether this observer should accumulate samples
+      with multiple calls to observe. Defaults to False.
+    :param float curvature_radius: Detectors in real bolometer cameras may
+      have curved corners due to machining limitations. This parameter species
+      the corner radius.
+
+    :ivar Vector3D normal_vector: The normal vector of the detector constructed from
+      the cross product of the x and y basis vectors.
+    :ivar float width: The extent of the bolometer foil in the basis_x direction.
+    :ivar float height: The extent of the bolometer foil in the basis_y direction.
+    :ivar array pixels_as_foils: A 2D array of pixels as individual BolometerFoil objects,
+      useful for calling BolometerFoil methods on each pixel (e.g. sightline tracing).
+      The array is indexed by (column, row).
+    :ivar Vector3D sightline_vectors: A 2D array of vectors that point from the centre
+      of each pixel on the foil detector to the centre of the slit. Defines the
+      effective sightline vectors of the pixels of the detector. The list is indexed by
+      (column, row).
+
+    .. code-block:: pycon
+
+       >>> from raysect.core import Point3D, Vector3D
+       >>> from raysect.optical import World
+       >>> from cherab.tools.observers import BolometerIRVB
+       >>>
+       >>> world = World()
+       >>>
+       >>> # construct basis vectors
+       >>> basis_x = Vector3D(1, 0, 0)
+       >>> basis_y = Vector3D(0, 1, 0)
+       >>> basis_z = Vector3D(0, 0, 1)
+       >>>
+       >>> # specify a detector, you need already created slit and camera objects
+       >>> width = 0.0025
+       >>> pixels = (10, 20)
+       >>> centre_point = Point3D(0, 0, -0.08)
+       >>> detector = BolometerIRVB("ch#1", centre_point, basis_x, basis_y, width, pixels, slit, parent=camera)
+    """
+
+    def __init__(self, detector_id, centre_point, basis_x, basis_y, width, pixels, slit,
+                 parent=None, units="Power", accumulate=False, curvature_radius=0):
+
+        # perform validation of input parameters
+
+        if not isinstance(width, (float, int)):
```

Comment:
Prescribing a type is not the best way to regularise data in python. Convert the type and then validate once it is in a known type. e.g. what would happen it a user had an np.float64? They would be forced to convert to a python float despite both being conceptually valid. Do the following (or equivalent)

```
# explictly convert type to type with known properties
width = float(width)

# tests follow with known type
if BLAH......
```

Reply:
Done.